json_schema/core/file/file_core.json

```
@@ -49,6 +49,17 @@
             "type": "string",
             "user_friendly": "Checksum",
             "example": "e09a986c2e630130b1849d4bf9a94c06"
-	    }
+	    },
+        "file_source": {
+            "description": "The source of the file. This is typically an organisation, person or dedicated process.",
+            "type": "string",
+            "enum":[
+                "DCP",
+                "Contributor"
+            ],
```

```suggestion
            "enum":[
                "DCP/2 Analysis",
                "Contributor",
                "Array Express",
                "GEO",
                "SCEA",
                "SCP",
                "DCP/1 Matrix Service",
                "LungMAP",
                "Zenodo",
                "Publication",
            ],
```

to be consistent with the set of sources that we currently track (https://github.com/DataBiosphere/azul/blob/6b45e0e30fbe90c7b3944ee691c95328c7883ab2/src/azul/plugins/metadata/hca/transform.py#L271).

It would be undesirable to be less informative than the current "hacky" mechanism.

---

Might also want to add `"HCA Release"`

---

@hannes-ucsc it might be nice to also represent if a file source type is "internal" or "external". We could do this by making this an enum of dictionaries instead of strings, with an added property called "type". This would be of use in simplifying to users the distinction between DCP and CGM matrices.

---

I did some searching but I can't find anything about having an enum of objects in json schema. Or are you suggesting an additional field in the schema? If it is a 'nice to have' would it be possible to make a ticket on the repo with your proposal and we will prioritise it. Thanks!

---

@hannes-ucsc I have incorporated the enums as requested in the PR I am making with all feedback from the review process (https://github.com/HumanCellAtlas/metadata-schema/pull/1381). We'll wait until test data has been reviewed and all expected changes are made in this branch before internal review, then squash and merge to minimise commits, then re-publish the schemas on staging.